### ISSUE TITLE
[MacOS Accessibility] - onDidGainAccessibilityFocus does not trigger on voiceover navigation focus.

### ISSUE THREAD

**Issue by magicwhale** [flutter/flutter#186271]
### Steps to reproduce

Add onDidGainAccessibilityFocus to a flutter component, use VoiceOver navigation on Mac to navigate to the component. Any code in onDidGainAccessibilityFocus does not trigger even though the component gained accessibility focus.

### Expected results

When VoiceOver navigation focuses on a component with onDidGainAccessibilityFocus, the callback should be executed.

### Actual results

Nothing happens when accessibility focus is gained.

### Code sample

https://dartpad.dev/?id=323157a85800bc22169ac8a75c7a4c3b

### Screenshots or Video

https://github.com/user-attachments/assets/ab073458-939b-4041-822a-cbbb2eb3e19e

### Logs

There are no logs, even though you would expect one from the print statement added to the onDidGainAccessibilityFocus callback in the example.

### Flutter Doctor output

N/A, using dartpad

**Comment by Istiak-Ahmed78** [comment]
It looks like it's Mac specific. I recommend to mention that on the title

**Comment by Istiak-Ahmed78** [comment]
I visited the dartpad where you shared the sample code. But I found this:
<img width="320" height="594" alt="Image" src="https://github.com/user-attachments/assets/e8630648-9647-405d-8989-47e115f3dcb4" />

Can you directly share the sample code in a code block? That may be more professional.

**Comment by saurabh-mirajkar** [comment]
@magicwhale Thanks for the detailed report. Labeling for `team-accessibility` for further investigation.

**Comment by magicwhale** [comment]
Here's the code from the dartpad:
```
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(colorSchemeSeed: Colors.blue),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  final String title;

  const MyHomePage({super.key, required this.title});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: Center(
        child: Column(
          mainAxisAlignment: .center,
          children: [
            const Text('You have pushed the button this many times:'),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
          ],
        ),
      ),
      floatingActionButton: Semantics(
        focusable: true,
        onDidGainAccessibilityFocus: () {
          print('gained accessibility focus!');
        },
        onFocus: () {
          print('gained focus.');
        },
        child: FloatingActionButton(
          onPressed: _incrementCounter,
          tooltip: 'Increment',
          child: const Icon(Icons.add),
        ),
      ),
    );
  }
}
```

**Comment by magicwhale** [comment]
dartpad link should also be working now!

**Comment by chunhtai** [comment]
I think the issue is because FloatingActionButton creates its own node, and the semantics widget wrap around it create a parent node wrapping the button. To make sure it gets fired, you should use a MergeSemantics to wrap around the whole thing to merge everything into one node
